samples/sample_multi_transcode/src/sample_multi_transcode.cpp

```diff
@@ -717,6 +717,12 @@ mfxStatus Launcher::QueryAdapters()
     mfxStatus sts = MFXQueryAdaptersNumber(&num_adapters_available);
     MFX_CHECK_STS(sts);
 
+    // no adapters on the machine, able to use software implementation
+    if (!num_adapters_available)
+    {
+        return MFX_ERR_NONE;
+    }
+
```

Comment:
Based on your comment, seems that we should add a check for implementation type (hw/sw).

Reply:
@AntonGrishin , ok, let’s fail more gracefully than just `MFX_ERR_UNSUPPORTED` from `InitEx()`.